src/views/course_pairings.vue

```diff
@@ -0,0 +1,239 @@
+
+export default {
+    data: function() {
+        return {
+            table: null,
+            specify_term: "",
+            column: "Instructor",
+            root: "table-format",
+            specify_course: "",
+            user_associations: [],
+            list_courses: [],
+            list_terms: [],
+            tableSettings: {
+                data: [],
+                stretchH: "all",
+                colHeaders: []
+            }
+        };
+    },
+    /**
+     * Function is called when page is create to get course pairings from API.
+     * Data is used to populate the spreadsheet and options for select
+     */
+    created: function() {
+        this.getData();
+    },
+
+    watch: {
+        /**
+         * Function is called when specify_course is modified. The data is used
+         * to populate the page with the specified course.
+         */
+        specify_course: function(oldVal, newVal) {
+            this.getData();
+        },
+
+        /**
+         * Function is called when specify_term is modified. The data is used
+         * to populate the page with the specified term.
+         */
+        specify_term: function(oldVal, newVal) {
+            this.getData();
+        },
+
+        /**
+         * Function is called when column is modified. The data is used
+         * to populate the page with the specified column.
+         */
+        column: function(oldVal, newVal) {
+            this.getData();
+        }
+    },
+    methods: {
+        /**
+         * Function gets course pairing data from API for the specified
+         * course, term and column and calls parseData to parse retuned data.
+         */
+        getData: function() {
+            // Converting to parameter enums
+            if (this.column == "TA") {
+                var column_name = "ta";
+            } else {
+                var column_name = "instructor";
+            }
+
+            var course_where = "";
+            var term_where = "";
+            if (this.specify_term != "") {
+                term_where = `&term=${this.specify_term}`;
+            }
+            if (this.specify_course != "") {
+                course_where = `&course_code=${this.specify_course}`;
+            }
+
+            var url =
+                `get_info.php?what=course_pairings&column=${column_name}&user_id` +
+                `=${this.$route.params.user_id}${term_where}${course_where}`;
+
+            fetch(url)
+                .then(res => res.json())
+                .then(data => this.parseData(data))
+                .catch(err => {
+                    this.$emit("error", err.toString());
+                });
+        },
+
+        /**
+         * Function parses data from API call that will be rendered on the
+         * spreadsheet and select component.
+         */
+        parseData: function(data) {
+            var pairings = {};
+            for (var index = 0; index < data.DATA.length; index++) {
```

Comment:
Why not
```
for (let elm of data.DATA) { ...
```
That would clean up the code a lot

Reply:
I use the index value within the for loop.